### PR TITLE
Move the type information from the names into metadata.

### LINKED ISSUE #561: Remove `BEL_FF` (etc) naming
A lot of the objects were named `BEL_FF` and `BLK_BB`. This was originally because we didn't have any way to attach metadata to the objects, so it was encoded as part of the name.

Now we have metadata support, we should move the info into metadata and go back to having human readable + understandable names.

### PR DESCRIPTION
**Pull request by mithro**
This metadata was originally encoded in the name as there was nowhere else to store it. Now that VPR has metadata support, we should store the information there.

Fixes #561.

### CODE REVIEW

**Comment by mithro** [comment]
@acomodi + @kgugala - FYI

File: ice40/cells/io_local/io_local.pb_type.xml [hunk 1]
```diff
@@ -17,28 +17,32 @@
...
+  <direct name="io_global_inclk1"  input="IO_LOCAL.io_global_inclk"  output="IO[1].INCLK"        />
+  <direct name="io_global_outclk0" input="IO_LOCAL.io_global_outclk" output="IO[0].OUTCLK"       />
+  <direct name="io_global_outclk1" input="IO_LOCAL.io_global_outclk" output="IO[1].OUTCLK"       />
+  <direct name="io_global_latch0"  input="IO_LOCAL.io_global_latch"  output="IO[0].LATCH"        />
+  <direct name="io_global_latch1"  input="IO_LOCAL.io_global_latch"  output="IO[1].LATCH"        />
  </interconnect>
+ <metadata>
+  <meta name="type">block</meta>
+  <meta name="subtype">ignore</meta>
+ </metadata>
```
Comment (acomodi) [inline ice40/cells/io_local/io_local.pb_type.xml]: This new metadata are not handled by genfasm right now, right? Moreover, what is the purpose of the `subtype` meta?

File: utils/lib/mux.py [hunk 2]
```diff
@@ -123,19 +123,19 @@ def pb_type_xml(mux_type, mux_name, pins, subckt=None, num_pb=1, comment=""):
 
     if mux_type == MuxType.LOGIC:
         if '-' not in mux_name:
-            mux_name = 'BEL_MX-' + mux_name
+            mux_name = '' + mux_name
```
Comment (acomodi) [inline utils/lib/mux.py]: Why is this needed?

File: utils/lib/mux.py [hunk 3]
```diff
@@ -123,19 +123,19 @@ def pb_type_xml(mux_type, mux_name, pins, subckt=None, num_pb=1, comment=""):
...
         else:
             assert mux_name.startswith(
-                'BEL_MX-'
+                ''
```
Comment (acomodi) [inline utils/lib/mux.py]: does this assertion make sense now that prefixes from the names are gone?

File: utils/vlog/vlog_to_pbtype.py [hunk 4]
```diff
@@ -125,18 +125,18 @@ def mod_pb_name(mod):
...
     elif mod_cls == "lut":
-        return "BEL_LT-" + mod.name
+        return mod.name
     elif is_blackbox and not has_modes:
-        return "BEL_BB-" + mod.name
+        return mod.name
     else:
         #TODO: other types
-        return "BLK_IG-" + mod.name
+        return mod.name
```
Comment (acomodi) [inline utils/vlog/vlog_to_pbtype.py]: I guess this function is not needed anymore and could be removed

File: xc7/fasm2bels/net_map.py [hunk 5]
```diff
@@ -37,7 +37,7 @@ def create_net_list(conn, graph, route_file):
         pin_name = graph.pin_ptc_to_name_map[(gridloc.block_type_id, node.ptc)]
 
         # Do not add synthetic nets to map.
-        if pin_name.startswith('BLK_SY-'):
+        if pin_name.startswith(''):
```
Comment (acomodi) [inline xc7/fasm2bels/net_map.py]: Is this correct?

File: xc7/utils/prjxray_arch_import.py [hunk 6]
```diff
@@ -311,7 +311,7 @@ def main():
             continue
         elif gridinfo.tile_type in tile_types:
             # We want to import this tile type.
-            vpr_tile_type = 'BLK_TI-{}'.format(gridinfo.tile_type)
+            vpr_tile_type = '{}'.format(gridinfo.tile_type)
```
Comment (acomodi) [inline xc7/utils/prjxray_arch_import.py]: Why not just `vpr_tile_type = gridinfo.tile_type`?

File: xc7/utils/prjxray_arch_import.py [hunk 7]
```diff
@@ -473,9 +473,9 @@ def main():
                         direct['x_offset'], direct['y_offset']
                     ),
                 'from_pin':
-                    'BLK_TI-' + direct['from_pin'],
+                    '' + direct['from_pin'],
                 'to_pin':
-                    'BLK_TI-' + direct['to_pin'],
+                    '' + direct['to_pin'],
```
Comment (acomodi) [inline xc7/utils/prjxray_arch_import.py]: leave only `direct['to/from pin']`

File: xc7/utils/prjxray_generate_dummy_site.py [hunk 8]
```diff
@@ -58,7 +58,7 @@ def main():
     pb_type_xml = ET.Element(
         'pb_type',
         {
-            'name': 'BLK_DU-{}'.format(args.site_type),
+            'name': '{}'.format(args.site_type),
```
Comment (acomodi) [inline xc7/utils/prjxray_generate_dummy_site.py]: remove `'{}'.format`

File: xc7/utils/prjxray_routing_import.py [hunk 9]
```diff
@@ -97,7 +97,7 @@ def import_graph_nodes(conn, graph, node_mapping):
             (gridloc.block_type_id, node.loc.ptc)]
 
         # Synthetic blocks are handled below.
-        if pin_name.startswith('BLK_SY-'):
+        if pin_name.startswith(''):
```
Comment (acomodi) [inline xc7/utils/prjxray_routing_import.py]: Is this correct?

File: xc7/utils/prjxray_tile_import.py [hunk 10]
```diff
@@ -244,7 +244,7 @@ def add_direct(xml, input, output):
             }
         )
 
-    tile_name = "BLK_TI-{}".format(args.tile)
+    tile_name = "{}".format(args.tile)
```
Comment (acomodi) [inline xc7/utils/prjxray_tile_import.py]: remove `'{}'.format`?

File: ice40/cells/io_local/io_local.pb_type.xml (hunk 1 above)
Comment (mithro) [reply ice40/cells/io_local/io_local.pb_type.xml]: They are documentation. In the future it might be used for GUI stuff.

File: xc7/fasm2bels/net_map.py (hunk 5 above)
Comment (mithro) [reply xc7/fasm2bels/net_map.py]: Your looking at an outdated version?

File: xc7/utils/prjxray_routing_import.py (hunk 9 above)
Comment (mithro) [reply xc7/utils/prjxray_routing_import.py]: Your looking at an old commit.

File: xc7/fasm2bels/net_map.py (hunk 5 above)
Comment (acomodi) [reply xc7/fasm2bels/net_map.py]: Yes, my bad

File: utils/lib/mux.py (hunk 2 above)
Comment (mithro) [reply utils/lib/mux.py]: It is not any longer.

Comment (mithro) [reply utils/lib/mux.py]: Its not, removed.

File: utils/vlog/vlog_to_pbtype.py (hunk 4 above)
Comment (mithro) [reply utils/vlog/vlog_to_pbtype.py]: Nope!

File: xc7/utils/prjxray_tile_import.py (hunk 10 above)
Comment (mithro) [reply xc7/utils/prjxray_tile_import.py]: Done.

File: xc7/utils/prjxray_arch_import.py (hunk 6 above)
Comment (mithro) [reply xc7/utils/prjxray_arch_import.py]: Done.

File: utils/lib/mux.py (hunk 3 above)
Comment (mithro) [reply utils/lib/mux.py]: Nope!

**Comment by mithro** [comment]
FYI @acomodi - I squashed the comments.

File: utils/vlog/vlog_to_pbtype.py (hunk 4 above)
Comment (acomodi) [reply utils/vlog/vlog_to_pbtype.py]: I am still unsure why this is useful as in all cases `mod.name` is returned. Is that because it still is incomplete (e.g. the TODO)?

**Comment by acomodi** [comment]
> FYI @acomodi - I squashed the comments.

Ok, LGTM. I would say it can be merged after CI is green.
